prog/test/github_runner.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require "octokit"
+require "yaml"
+
+class Prog::Test::GithubRunner < Prog::Test::Base
+  FAIL_CONCLUSIONS = ["action_required", "cancelled", "failure", "skipped", "stale", "timed_out"]
+  IN_PROGRESS_CONCLUSIONS = ["in_progress", "queued", "requested", "waiting", "pending", "neutral"]
+
+  def self.assemble(vm_host_id, test_cases)
+    github_service_project = Project.create(name: "Github Runner Service Project") { _1.id = Config.github_runner_service_project_id }
+    github_service_project.associate_with_project(github_service_project)
+
+    github_test_project = Project.create_with_id(name: "Github Runner Test Project")
+    GithubInstallation.create_with_id(
+      installation_id: Config.github_test_installation_id,
+      name: "GithubTestInstallationName",
+      type: "GithubTestInstallationType",
+      project_id: github_test_project.id
+    )
+    github_test_project.associate_with_project(github_test_project)
+
+    Strand.create_with_id(
+      prog: "Test::GithubRunnerGroup",
+      label: "start",
+      stack: [{
+        "vm_host_id" => vm_host_id,
+        "test_cases" => test_cases,
+        "github_service_project_id" => github_service_project.id,
+        "github_test_project_id" => github_test_project.id
+      }]
+    )
+  end
+
+  label def start
+    hop_download_boot_images
+  end
+
+  label def download_boot_images
+    frame["test_cases"].each do |test_case|
+      bud Prog::DownloadBootImage, {"subject_id" => vm_host_id, "image_name" => tests[test_case]["image_name"]}
+    end
+
+    hop_wait_download_boot_images
+  end
+
+  label def wait_download_boot_images
+    reap
+    hop_trigger_test_runs if leaf?
+    donate
+  end
+
+  label def trigger_test_runs
+    test_runs.each do |test_run|
+      trigger_test_run(test_run["repo_name"], test_run["workflow_name"], test_run["branch_name"])
+    end
+
+    # To make sure that test runs are triggered
+    sleep 30
+
+    hop_check_test_runs
+  end
+
+  label def check_test_runs
+    test_runs.each do |test_run|
+      conclusion = test_run_conclusion(test_run["repo_name"], test_run["workflow_name"], test_run["branch_name"])
+      if FAIL_CONCLUSIONS.include?(conclusion)
+        fail_test "Test run for #{test_run["repo_name"]}, #{test_run["workflow_name"]}, #{test_run["branch_name"]} failed with conclusion #{conclusion}"
+      elsif IN_PROGRESS_CONCLUSIONS.include?(conclusion) || conclusion.nil?
+        nap 15
+      end
+    end
+
+    hop_finish
+  end
+
+  label def finish
+    cleanup
+    pop "GithubRunnerGroup tests are finished!"
+  end
+
+  label def failed
+    cleanup
+    nap 15
+  end
+
+  def trigger_test_run(repo_name, workflow_name, branch_name)
+    url = "repos/#{repo_name}/actions/workflows/#{workflow_name}/dispatches"
```

Comment:
nit: Octokit.rb has a method for it
https://github.com/octokit/octokit.rb/blob/c976852a4b54a967f26b5b943f7bc983ad2e3514/lib/octokit/client/actions_workflows.rb#L41